nucliadb_vectors/src/vectors/data_point/mod.rs

```diff
@@ -278,9 +278,14 @@ impl DataPoint {
         let node_producers = operants
             .iter()
             .map(|dp| ((dp.0, Node), dp.1.nodes.as_ref()));
-        key_value::merge(&mut nodes, node_producers.collect())?;
-        nodes.flush()?;
+
+        let mut node_buffer = BufWriter::new(&mut nodes);
+        key_value::merge(&mut node_buffer, node_producers.collect())?;
+        node_buffer.flush()?;
+        std::mem::drop(node_buffer);
```

Comment:
Maybe it is a bit more idiomatic to use nested scope instead of manually call the `drop` function :
```rust
{
        let mut node_buffer = BufWriter::new(&mut nodes);
        key_value::merge(&mut node_buffer, node_producers.collect())?;
        node_buffer.flush()?;
}
```

Reply:
Done at [b5a8af7](https://github.com/nuclia/nucliadb/pull/455/commits/b5a8af788ee8915678e1d15695fc046e496ab20c)